src/contexts/providers.context.tsx

```diff
@@ -49,35 +61,57 @@ const ProvidersProvider: FC = (props) => {
       setAccount({ status: "loading" });
       switch (walletName) {
         case WalletName.METAMASK: {
-          if (window.ethereum && window.ethereum.isMetaMask) {
-            const web3Provider = new Web3Provider(window.ethereum);
-            return getConnectedAccounts(web3Provider)
-              .then((accounts) => {
-                setConnectedProvider(web3Provider);
-                setAccount({ status: "successful", data: accounts[0] });
-              })
-              .catch((error) => {
-                if (isMetamaskUserRejectedRequestError(error)) {
-                  setAccount({ status: "pending" });
-                } else {
-                  void parseError(error).then((parsed) => {
-                    openSnackbar({
-                      type: "error",
-                      parsed,
-                    });
-                  });
-                }
-              });
-          } else {
+          if (env === undefined) {
             return setAccount({
               status: "failed",
-              error: `We cannot detect your wallet. Make sure the ${WalletName.METAMASK} extension is installed and active in your browser.`,
+              error: "The env has not been initialized correctly",
             });
           }
+
+          try {
+            if (window.ethereum && window.ethereum.isMetaMask) {
+              const web3Provider = new Web3Provider(window.ethereum);
+              const requestedNetwork = await web3Provider.getNetwork();
+              const supportedNetworks = await Promise.all(
+                env.chains.map((chain) => chain.provider.getNetwork())
+              );
+              const requestedChainId = requestedNetwork.chainId;
+              const supportedChainIds = supportedNetworks.map((network) => network.chainId);
+
+              if (!supportedChainIds.includes(requestedChainId)) {
+                return setAccount({
+                  status: "failed",
+                  error: "Switch your network to Ethereum or Polygon Hermez to continue",
+                });
+              }
```

Comment:
https://user-images.githubusercontent.com/15896177/165972428-e3c56c69-9794-4f81-8a7b-69270235403f.mp4

Reply:
I think that this is related to the latest PR merged. I'll create an issue to take a more detailed look to this in another PR